src/state/opamSysInteract.ml

```diff
@@ -237,6 +239,65 @@ module Cygwin = struct
   let cygroot config = get_opt (cygroot_opt config)
   let cygsetup config = get_opt (cygsetup_opt config)
 
+  let download_setupexe dst =
+    let open OpamProcess.Job.Op in
+    OpamFilename.with_tmp_dir_job @@ fun dir ->
+    OpamDownload.download ~overwrite:true url_setupexe dir @@| fun filename ->
+    (* XXX check signature *)
```

Comment:
In the meantime, as a cheap workaround, we could just display the sha512 and ask for confirmation.
Apparently cygwin postes those at https://cygwin.com/sha512.sum

Reply:
```
 let download_setupexe dst =
    let open OpamProcess.Job.Op in
    OpamFilename.with_tmp_dir_job @@ fun dir ->
    OpamDownload.download ~overwrite:true url_setupexe_sha512 dir @@| fun file ->
    let checksum =
      let content = OpamFilename.read file in
      let re =
        Re.(compile @@ seq [
            group @@ repn
              (alt [ digit ; rg 'A' 'F'; rg 'a' 'f' ]) 128 (Some 128);
            rep space;
            str "setup-x86_64.exe"
          ])
      in
      try Some (OpamHash.sha512 Re.(Group.get (exec re content) 1))
      with Not_found -> None
    in
    OpamDownload.download_as ~overwrite:true ?checksum url_setupexe dst